data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/tracing/kafka/ConsumerTracer.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright © 2018 Knative Authors (knative-dev@googlegroups.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * This piece of code is inspired from vert-x3/vertx-kafka-client project.
+ * The original source code can be found here: https://github.com/vert-x3/vertx-kafka-client
+ */
```

Comment:
perhaps me, but I'd add this on the comment of the `clazz`, and also point a link of the `sha` in GIT, that matches this file, at the time of copying.